lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -167,6 +177,30 @@ RTLExternModuleOp ESIRTLBuilder::declareStage() {
   return declaredStage;
 }
 
+/// Write an 'ExternModuleOp' to use a hand-coded SystemVerilog module. Said
+/// module contains a bi-directional Cosimulation DPI interface with valid/ready
+/// semantics.
+RTLExternModuleOp ESIRTLBuilder::declareCosimEndpoint() {
+  if (declaredCosimEndpoint)
+    return declaredCosimEndpoint;
+  auto name = StringAttr::get("Cosim_Endpoint", getContext());
+  // Since this module has parameterized widths on the a input and x output,
+  // give the extern declation a None type since nothing else makes sense.
+  // Will be refining this when we decide how to better handle parameterized
+  // types and ops.
+  ModulePortInfo ports[] = {
+      {clk, PortDirection::INPUT, getI1Type(), 0},
+      {rstn, PortDirection::INPUT, getI1Type(), 1},
+      {dataOutValid, PortDirection::OUTPUT, getI1Type(), 0},
+      {dataOutReady, PortDirection::INPUT, getI1Type(), 2},
+      {dataOut, PortDirection::OUTPUT, getNoneType(), 1},
+      {dataInValid, PortDirection::INPUT, getI1Type(), 3},
+      {dataInReady, PortDirection::OUTPUT, getI1Type(), 2},
+      {dataIn, PortDirection::INPUT, getNoneType(), 4}};
+  declaredStage = create<RTLExternModuleOp>(name, ports);
```

Comment:
I think this and the return on the next line need to use `declaredCosimEndpoint`?

Reply:
Good catch.